src/crustls.h

```diff
@@ -232,6 +232,24 @@ typedef struct rustls_slice_str rustls_slice_str;
  */
 typedef struct rustls_supported_ciphersuite rustls_supported_ciphersuite;
 
+/**
+ * A read-only view on a Rust `&str`. The contents are guaranteed to be valid
+ * UTF-8. As an additional guarantee on top of Rust's normal UTF-8 guarantee,
+ * a `rustls_str` is guaranteed not to contain internal NUL bytes, so it is
+ * safe to interpolate into a C string or compare using strncmp. Keep in mind
+ * that it is not NUL-terminated.
+ *
+ * The memory exposed is available as specified by the function
+ * using this in its signature. For instance, when this is a parameter to a
```

Comment:
> The memory exposed is available as specified by the function using this in its signature.

To be honest, I'm not sure how to interpret this if I'm calling this from C code.